### PR TITLE
schema allow both example and description simultaneously

### LINKED ISSUE #37: openapi: adding "description" removes "example"
The "description" and "example" seem to be mutually exclusive in rweb-generated openapi.

If I have a code like this:
```Rust
#[derive(Debug, Serialize, Deserialize, Schema)]
pub struct ExampleTest {
    #[schema(example = "10")]
    data: usize,
    #[schema(example = "\"Example for string values must be escaped like this\"")]
    something: String,
}
```
the generated openapi specification would have:
```json
              "schema": {
                "type": "object",
                "properties": {
                  "data": {
                    "type": "integer",
                    "example": 10
                  },
                  "something": {
                    "type": "string",
                    "example": "Example for string values must be escaped like this"
                  }
                }
              }
```
as we see, examples are provided for both fields.

But if we then add a rustdoc comment to the code:
```rust
#[derive(Debug, Serialize, Deserialize, Schema)]
pub struct ExampleTest {
    /// this is a description
    #[schema(example = "10")]
    data: usize,
    #[schema(example = "\"Example for string values must be escaped like this\"")]
    something: String,
}
```
then the generated openapi description will have:
```json
              "schema": {
                "type": "object",
                "properties": {
                  "data": {
                    "description": " this is a description",
                    "type": "integer"
                  },
                  "something": {
                    "type": "string",
                    "example": "Example for string values must be escaped like this"
                  }
                }
              }
```
As we see, the example section for the documented field is gone.

Thoughts?

### PR DESCRIPTION
**Pull request by E-gy**
TLDR: Fixes #37 

Currently, the only case that works was
```rust
#[derive(Schema)]
pub struct TestStruct {
    #[schema(example = "\"an example\"")]
    #[schema(description = "a description")]
    d2: String,
}
```

Now any permutation of doc/attributes works
```rust
#[derive(Schema)]
pub struct TestStruct {
    ///a description
    #[schema(example = "\"an example\"")]
    d1: String,
    #[schema(example = "\"an example\"")]
    #[schema(description = "a description")]
    d2: String,
    #[schema(example = "\"an example\"", description = "a description")]
    d3: String,
}
```

PS:
Once rust-lang/rust#54140 arrives, `schema` invalid usage errors can be more descriptive.

### CODE REVIEW

**Review by kdy1 — APPROVED** [review]
LGTM. Thanks!

**Comment by vn971** [comment]
Yaaay, thanks! :tada:

**Comment by vn971** [comment]
(I've actually switched to pure `warp`, so doesn't affect me, but nice to see it solved)
